### ISSUE TITLE
NullPointerException using model with HubNet

### ISSUE THREAD

**Issue by ScootCork** [NetLogo/NetLogo#847]
NetLogo threw this error at me when working with HubNet. I'm currently using NetLogo to create a game for students to play and am a little clueless about what is going wrong. Any help regarding suggestions for a workaround or a little explanation would be greatly appreciated! 

[EDIT]: I must say that I really enjoy working with NetLogo and that it works great in an educational setting. So my gratitude goes out to all who created and maintain this program! :)

The specific error happened with this line in the code:

ask companies with [ user-id = hubnet-message-source ] [...]

The error:

error (NullPointerException)
 while observer running ASK
  called by procedure LISTEN-CLIENTS (C:\Users\Jeroen\Desktop\hardcopy model\Model\hubnet.nls)
  called by procedure PLAY
  called by Button 'Play'

NetLogo is unable to supply you with more details about this error.  Please report the problem
at https://github.com/NetLogo/NetLogo/issues, or to bugs@ccl.northwestern.edu, and paste the
contents of this window into your report.

java.lang.NullPointerException
 at org.nlogo.prim._asm_procedurelistenclientscusersjeroendesktophardcopymodelmodelhubnetnls_equal_11.report(:1)
 at org.nlogo.nvm.Context.evaluateReporter(Context.java:227)
 at org.nlogo.prim._asm_procedurelistenclientscusersjeroendesktophardcopymodelmodelhubnetnls_ask_10.perform(:2)
 at org.nlogo.nvm.Context.stepConcurrent(Context.java:91)
 at org.nlogo.nvm.ConcurrentJob.step(ConcurrentJob.java:82)
 at org.nlogo.job.JobThread.org$nlogo$job$JobThread$$runPrimaryJobs(JobThread.scala:143)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply$mcV$sp(JobThread.scala:78)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:76)
 at org.nlogo.job.JobThread$$anonfun$run$1.apply(JobThread.scala:76)
 at scala.util.control.Exception$Catch.apply(Exception.scala:88)
 at org.nlogo.util.Exceptions$.handling(Exceptions.scala:41)
 at org.nlogo.job.JobThread.run(JobThread.scala:75)

NetLogo 5.2.0
main: org.nlogo.app.AppFrame
thread: JobThread
Java HotSpot(TM) Server VM 1.6.0_45 (Sun Microsystems Inc.; 1.6.0_45-b06)
operating system: Windows 8 6.2 (x86 processor)
Scala version 2.9.2
JOGL: (3D View not initialized)
OpenGL Graphics: (3D View not initialized)
model: gameV0.7.4

10:35:35.223 SwitchedTabsEvent (org.nlogo.app.Tabs) AWT-EventQueue-0
10:35:35.223 RuntimeErrorEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
10:35:35.223 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
10:35:35.223 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
10:35:35.223 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
10:35:35.223 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
10:35:35.223 PeriodicUpdateEvent (org.nlogo.app.App$$anon$1 (org.nlogo.window.GUIWorkspace)) AWT-EventQueue-0
10:35:35.096 AddJobEvent (org.nlogo.window.ButtonWidget) AWT-EventQueue-0
10:35:35.018 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0
10:35:35.018 InterfaceGlobalEvent (org.nlogo.app.InterfacePanel$2 (org.nlogo.window.SliderWidget)) AWT-EventQueue-0

**Comment by SethTisue** [comment]
probably #92 rearing its head yet again

**Comment by ScootCork** [comment]
Looking at the steps to reproduce in #92 I think something similar but different happened:
The server ran into a normal error with the clients connected (bug in my code). I then reset the server, changed some code and let the clients reconnect (not completely sure about the order I did those in, is a reset necessary after code changes?). A little while after the clients connected again I got the error mentioned above, this was not in the login phase.
